src/bus/index.js

```diff
@@ -12,13 +15,73 @@ class Bus {
    */
   constructor(sdk, request) {
     const baseUrl = `${sdk.config.audiences.bus.host}`;
-
+    const baseSocketUrl = `${sdk.config.audiences.bus.socket}`;
+    this._baseSocketUrl = baseSocketUrl;
     this._baseUrl = baseUrl;
     this._request = request;
     this._sdk = sdk;
+    this._sockets = {};
 
     this.channels = new Channels(sdk, request, baseUrl);
   }
+
+  /**
+   * Connects to the message bus via websocket.
+   * If a connection already exists for that organization id, the connection is returned, otherwise a new connection is created.
+   *
+   * @param {string} organizationId UUID corresponding with an organization
+   *
+   * @returns {Promise}
+   * @fulfill {Socket}
+   * @reject {object} The error event
+   *
+   * @example
+   * contxtSdk.bus.connect('4f0e51c6-728b-4892-9863-6d002e61204d')
+   * .then((socket) => {
+   *    console.log(socket);
+   * })
+   * .catch((event) => {
+   *    console.log(event);
```

Comment:
I'm not familiar with the `ws` library but should this be `err` or `error` similar to other places in the contxt sdk?

Reply:
Everything is event based in the websocket spec so even though it's an "error" it's still an event

Reply:
But I don't have anything against naming it `error`

Reply:
renamed `errorEvent`